sdk/ml/azure-ai-ml/azure/ai/ml/entities/_validation.py

```diff
@@ -483,35 +483,58 @@ def from_single_message(
         return obj
 
     @classmethod
-    def from_validation_error(cls, error: ValidationError, source_path: str = None):
+    def from_validation_error(cls, error: ValidationError, source_path: str = None, error_on_unknown_field=False):
         """Create a validation result from a ValidationError, which will be
         raised in marshmallow.Schema.load. Please use this function only for
         exception in loading file.
 
         param error: ValidationError raised by marshmallow.Schema.load.
+        type error: ValidationError
+        param error_on_unknown_field: whether to raise error if there
+        are unknown field diagnostics.
+        type error_on_unknown_field: bool
         """
-        obj = cls.from_validation_messages(error.messages, data=error.data)
+        obj = cls.from_validation_messages(
+            error.messages, data=error.data,
+            error_on_unknown_field=error_on_unknown_field
+        )
         if source_path:
             obj.resolve_location_for_diagnostics(source_path, resolve_value=True)
         return obj
 
     @classmethod
-    def from_validation_messages(cls, errors: typing.Dict, data: typing.Dict):
+    def from_validation_messages(
+        cls,
+        errors: typing.Dict,
+        data: typing.Dict,
+        error_on_unknown_field: bool = False
```

Comment:
better to have a '*' before 'error_on_unknown_field' I think, since it's a bool with default.

Reply:
good point!